src/main/java/com/derongan/minecraft/deeperworld/DeeperWorldExtensions.kt

```diff
@@ -0,0 +1,13 @@
+package com.derongan.minecraft.deeperworld
+
+import org.bukkit.entity.Entity
+
+internal fun Entity.getVehicleRecursive() : Entity? {
```

Comment:
I think you should be able to label this function as `tailrec` to get a small performance boost.